### ISSUE TITLE
SSH v2.99

### ISSUE THREAD

**Issue by swartzlib7** [phpseclib/phpseclib#1170]
I have a router reporting back SSH v2.99 which causes the SSH2.php file to abort.

When adding version 2.99 to the condition as follow all works:
if ($matches[3] != '1.99' && $matches[3] != '2.0' && $matches[3] != '2.99')

Not sure what the implication is when adding this in and if you could safely add it in?

Please let me know.
Stephen

**Comment by terrafrost** [comment]
Does it work with PuTTY? It's not immediately obvious upon quickly glancing at [PuTTY's source](https://github.com/github/putty/blob/master/ssh.c#L3307).

Technically it is malformed. Quoting https://tools.ietf.org/html/rfc4253#section-5.1 ,

>    Server implementations MAY support a configurable compatibility flag
>    that enables compatibility with old versions.  When this flag is on,
>    the server SHOULD identify its 'protoversion' as "1.99".  Clients
>    using protocol 2.0 MUST be able to identify this as identical to
>    "2.0".

**Comment by terrafrost** [comment]
It seems to work with PuTTY. To test I did `sudo nc -l 22` on my Linux and then used PuTTY to connect. PuTTY doesn't [send the identification string first](https://github.com/phpseclib/phpseclib/pull/1162) so I typed in `SSH-2.99-zzzz` in my Linux box and hit enter. PuTTY responded with `SSH-2.0-PuTTY-Release_0.63` and then sent the initial kex packet. So it seems like PuTTY likes it.

From https://github.com/github/putty/blob/2d0b2e97d00c5ca08a6cbd4a7d724b41e1222514/ssh.c#L3170 

```c
    /* Anything greater or equal to "1.99" means protocol 2 is supported. */
    s->proto2 = ssh_versioncmp(s->version, "1.99") >= 0;
```

I've gone ahead and made phpseclib behave similarly:

https://github.com/phpseclib/phpseclib/commit/9cbfd23cb65796e048558054d9b40f959ba7e4f8

Thanks!
